pkg/chrootarchive/archive_unix.go

```diff
@@ -107,12 +107,15 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 	w.Close()
 
 	if err := cmd.Wait(); err != nil {
+		errorOut := fmt.Errorf("processing tar file(%s): %w", output, err)
 		// when `xz -d -c -q | storage-untar ...` failed on storage-untar side,
 		// we need to exhaust `xz`'s output, otherwise the `xz` side will be
 		// pending on write pipe forever
-		io.Copy(io.Discard, decompressedArchive)
+		if _, err := io.Copy(io.Discard, decompressedArchive); err != nil {
+			return fmt.Errorf("%w and exhaust `xz`'s output failed: %w", errorOut, err)
```

Comment:
I think this would be rather a confusing output. e.g. `processing tar file(invalid header): exit status 1 and exhaust xz's output failed: I/O error`, it does not express the relationship of the two errors:
- “unpacking failed (error: …; output: …)”; and
- “exhausting input failed (error: …)”

---

Also note that `xz` in the above is an example, not an inherent component of the situation. (I’m not even sure that we _should_ have to exhaust the input, it seems to me that that’s what SIGPIPE / EPIPE is for, but I didn’t look into the details and that’s probably another PR.)